protocol/contracts/governance/TemplarMetadata.sol

```diff
@@ -0,0 +1,41 @@
+//SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "./Templar.sol";
+
+/**
+ * @title a contract recording the metadata for templars (ie roles etc).
+ */
+contract TemplarMetadata is AccessControl {
+
+    bytes32 public constant CAN_UPDATE = keccak256("CAN_UPDATE");
+
+     // Mapping from discordId ID to max temple role
+    mapping(uint256 => string) public templeRole;
+
+    /// @notice the NFT contract for templars
+    Templar public templars;
+
+    constructor(Templar _templars) {
+        templars = _templars;
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /**
+     * Set the (highest) temple role held by a templar with the given discordid
+     */
+    function setRole( 
+        uint256 discordId,
+        string calldata _templeRole
+    ) external onlyRole(CAN_UPDATE) {
+        templars.checkExists(discordId);
```

Comment:
I wonder if this should just return a bool such that the client can decide if it will revert or not

Reply:
It was a conscious decision so that there was a standard error that was thrown. 

But an addition function that returns a bool would be useful - will add.

Reply:
This is done.